lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py

```diff
@@ -236,10 +225,10 @@ def nacls_changed(nacl, client, module):
     nacl_id = nacl['NetworkAcls'][0]['NetworkAclId']
     nacl = describe_network_acl(client, module)
     entries = nacl['NetworkAcls'][0]['Entries']
-    tmp_egress = [entry for entry in entries if entry['Egress'] is True and DEFAULT_EGRESS != entry]
+    tmp_egress = [entry for entry in entries if entry['Egress'] is True and entry['RuleNumber'] < 32767]
     tmp_ingress = [entry for entry in entries if entry['Egress'] is False]
-    egress = [rule for rule in tmp_egress if DEFAULT_EGRESS != rule]
-    ingress = [rule for rule in tmp_ingress if DEFAULT_INGRESS != rule]
+    egress = [rule for rule in tmp_egress if rule['RuleNumber'] < 32767]
+    ingress = [rule for rule in tmp_ingress if rule['RuleNumber'] < 32767]
```

Comment:
Not sure why this module originally used `tmp_egress` and `tmp_ingress` lists as well as the `egress` and `ingress` lists. I think this could be simplified further to be just:

```
egress = [rule for rule in tmp_egress if rule['Egress'] is True and rule['RuleNumber'] < 32767]
ingress = [rule for rule in tmp_ingress if rule['Egress'] is False and rule['RuleNumber'] < 32767]
```

Or is there a reason the list needs to be filtered twice?

Reply:
Agreed. I was wondering the same thing. I will remove the tmp variables.

Reply:
I have made and tested the change to remove the tmp_egress and tmp_ingress variables.